pkg/prometheus/server/operator.go

```diff
@@ -1525,6 +1621,16 @@ func (c *Operator) createOrUpdateConfigurationSecret(ctx context.Context, p *mon
 	if err != nil {
 		return errors.Wrap(err, "selecting Probes failed")
 	}
+
+	var scrapeConfigs map[string]*monitoringv1alpha1.ScrapeConfig
+	if c.scrapeConfigSupported {
+		scrapeConfigs, err = resourceSelector.SelectScrapeConfigs()
+		if err != nil {
+			// ScrapeConfigs are still optional,
+			level.Error(c.logger).Log("msg", errors.Wrap(err, "selecting ScrapeConfigs failed"), "prometheus", p.Name, "namespace", p.Namespace)
+		}
```

Comment:
same comment as Simon previously but I think we would want to return the error here?

Reply:
25aa19594